### PR TITLE
Bundle ExcelJS, jsPDF, Roboto font and @mdi/font using npm instead of CDNs

### PR DESCRIPTION
**Pull request by quinn-dev**
Installs ExcelJS, jsPDf, the Roboto font and the MaterialDesignIcons as npm dependencies and removes references to external CDNs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,11 @@
     "lint": "vue-cli-service lint"
   },
   "dependencies": {
+    "@fontsource/roboto": "^4.5.0",
```
Comment (andrew-sayers) [inline package.json:11]: Should this go with the previous commit, rather than this one?

File: src/plugins/vuetify.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
+import '@mdi/font/css/materialdesignicons.css';
 import Vue from 'vue';
 import Vuetify from 'vuetify/lib/framework';
 
 Vue.use(Vuetify);
 
 export default new Vuetify({
-    theme: { dark: true },
+    theme: { dark: true }
```
Comment (andrew-sayers) [inline src/plugins/vuetify.js:8]: Trailing commas should only be removed when you want to signal an intention never to add a line after this one, which isn't the case here.  (This is a personal opinion, but a strong one I've never heard a decent counter-argument to)

Imagine you wanted to add another config option after `theme`.  You would need to add a trailing comma, which creates an unnecessary extra line to review.

Also, not relevant here, but if you ever need to auto-generate code, it's much easier to print a list of values if you don't have to worry about that special case.

File: src/views/Info.vue [hunk 3]
```diff
@@ -283,7 +285,7 @@
 
          generate_predictions() {
              window.prediction_spreadsheet(
-                 new window.ExcelJS.Workbook(),
+                 new ExcelJS.Workbook(),
```
Comment (andrew-sayers) [inline src/views/Info.vue:288]: a) this change isn't justified by the commit message, or the associated PR.  If it's important, it should go in a separate PR where we can discuss it
b) this `window`, and the one below, are there for ease of copy/pasting with code compiled with closure.  There are other solutions, but fair warning - it's a topic that will take a lot of time without generating much value

File: src/App.vue [hunk 4]
```diff
@@ -105,11 +105,13 @@
...
- import router from "./router/index.js";
+import "@fontsource/roboto";
 
- import diary_manager from "@/diary_manager.js";
+import router from "./router/index.js";
 
- import { PROJECT_URL } from "@/constants.js";
+import diary_manager from "@/diary_manager.js";
+
+import { PROJECT_URL } from "@/constants.js";
```
Comment (andrew-sayers) [inline src/App.vue]: a) if you want to make whitespace changes associated with a line, put it in a separate commit - right now I can't easily tell what this is trying to achieve
b) either use the same indentation as the surrounding lines, or introduce a PR that changes it across the project (preferably) or file (at minimum).  Consistency is more important than anything
c) discussions about indentation tend to get quite serious, so if this is something you feel strongly about, you might want to create an issue before spending too much time on a specific solution

**Review by andrew-sayers — COMMENTED** [review]
Here are some general issues of code style that will need to be addressed before this can go through.  I'll try to be clear which are generally considered best practice vs. which ones I just happen to have strong opinions about.

First, this would be better if it was split into more, smaller commits (admittedly I'm also trying to get out of that bad habit).  Most people find smaller commits easier to review, because they can look at the commit message and see how each line pertains to the stated goal.  As a rule of thumb, if a change isn't justified by the commit message, or if the message contains the word "and", it's usually time to split out a separate commit.  Some specific review comments highlight that issue.

Second, it seems like much of this follows (or should follow) standard instructions.  I haven't heard much discussion about best practice for that sort of thing, but personally I would prefer:
* each commit to reference the associated instructions (e.g. appending a line like "As discussed in https://vuetifyjs.com/en/features/icon-fonts/" to the message)
* commands (e.g. `yarn add <package>`) to go in their own commit
* commits associated with a command to include the text of that command in the commit message

[This library commit](https://github.com/sleepdiary/library/commit/16a136b036b79ac636788b6a1b06b41d087afdf6) is a particularly good example of why the last two are important.  I'll need to make a similar commit next year, and in more repositories.  Knowing the command I used last time, and the effects of that command, I can create and verify an analogous commit much more easily.

Finally, I could be wrong but this PR seems to focus on committing one file at a time rather than individual lines.  Git has [excellent tools for splitting commits by line](https://git-scm.com/book/en/v2/Git-Tools-Interactive-Staging), which make it easy to produce more granular commits.

File: src/App.vue (hunk 4 above)
Comment (quinn-dev) [reply src/App.vue]: a) Good point
b) / c) I'll respond to this in a separate discussion.

File: src/views/Info.vue (hunk 3 above)
Comment (quinn-dev) [reply src/views/Info.vue:288]: a) I thought the change is self evident - since the library isn't loaded into the global scope but as a module instead, it can no longer be referenced using `window.ExcelJS`.
b) I'm unsure what you mean by ease of copy/pasting with code compiled with closure.

Comment (andrew-sayers) [reply src/views/Info.vue:288]: Ah yes, my mistake.  The [Closure Compiler](https://developers.google.com/closure/compiler) requires some tricks to make it work, and this code is similar to some code in a repo which uses that compiler, but that's a problem for another day.
